docs/proposals/2017-04-27 Text Formatting.md

```diff
@@ -0,0 +1,239 @@
+
+Example:
+```
+  {
+    "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAA",
+    "type": "text",
+    "text": "The text attribute should contain plain text content only.",
+    "selections": [ ... ]
+    "content": "The <b>text</b> attribute should contain <i>plain text</i> content only."
+  }
+```
+
+### `selection`s describe formatting for specific segments of a text string
+
+Each `selection` object defines a part of a text string and what formatting attributes should be applied to it. An idea borrowed from Apple and Twitter.
+
+* a required `start` and `length` integer properties define sub-string range,
+* an optional `attributes` property with array of `attribute` objects.
+
+Example:
+```
+    "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAA",
+    "text": "The text attribute should contain plain text content only.",
+    "selections": [
+      {
+        "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAB",
+        "type": "selection",
+        "start": 4,
+        "length": 4,
+        "attributes": [ ... ]
+      },
+      { ... }
+    ]
+    "content": "The <b>text</b> attribute should contain <i>plain text</i> content only."
+```
+
+### `attribute`s are modifications applied to the text string or selections
+
+Each attribute element modifies the formatting of the text it is applied to. Initial list of supported attribute types shall be:
+
+* strong,
+* emphasis,
+* strike,
+* highlight,
+* superscript,
+* subscript,
+* smallcaps,
+* dropcaps,
+* link.
+
+Example:
+```
+        "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAB",
+        "type": "selection",
+        "start": 4,
+        "length": 4,
+        "attributes": [
+          {
+            "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAC",
+            "type": "strong"
+          }
+        ]
+```
+
+Additionally, `link` attribute must have `url` property pointing to the linked content. Example:
+
+```
+        "attributes": [
+          {
+            "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAC",
+            "type": "link"
+            "url": "https://www.washingtonpost.com/..."
+          }
+        ]
+```
+
+This initial list of attributes can be extended at a later versions of the specification with additional attributes, or with additional properties on existing attributes, if such need arises. For example, `strong` attribute could be extended to include `weight` property:
+
+```
+        "attributes": [
+          {
+            "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAC",
+            "type": "strong",
+            "weight": 500
+          }
+        ]
+```
+
+### `content` is an HTML representation of the canonical `text` property
+
+Each attributed text object can have an optional `content` element which contains HTML representation of that `text`. It is expected that, on input:
+
+* `text`, and `attributes`, and `selections` properties are present, or recreated from the `content` property during import;
+* `content` property is omitted, or contains only allowed subset of the HTML;
+
+On output:
```

Comment:
Does this mean that both `text` and `content` are used to display text in a published story depending on the platform the story is being read on (Apple News, AMP, a web browser)?